### ISSUE TITLE
timeout macro for tests which return Result

### ISSUE THREAD

**Issue by kingsimmy** [becheran/ntest#5]
Currently the timeout macro does not work when you add the timeout attribute to a unit test which returns a result.

e.g.

```rust
#[test]
#[timeout(100)]
fn no_timeout() -> Result<(), String> {
    let fifty_millis = time::Duration::from_millis(50);
    thread::sleep(fifty_millis);
    Ok(())
}
```
Would it be possible for you to enhance the macro so that it can wrap such a function and return the correct type

**Comment by becheran** [comment]
Ah cool. This rust 2018 feature of return values in test didn't exist when I created the lib. I will have a look.

**Comment by becheran** [comment]
I released version 0.6.0 including this new feature/enhancement. Hope you find it useful.

**Comment by kingsimmy** [comment]
Thank you. This works very well now.
